src/app/core/utilities/equals.decorators.ts

```diff
@@ -0,0 +1,58 @@
+import { isEmpty } from '../../shared/empty.util';
+import { GenericConstructor } from '../shared/generic-constructor';
+import { EquatableObject } from './equatable';
+
+const excludedFromEquals = new Map();
+const fieldsForEqualsMap = new Map();
+
+export function inheritEquatable(parentCo: GenericConstructor<EquatableObject<any>>) {
+  return function decorator(childCo: GenericConstructor<EquatableObject<any>>) {
+    const parentExcludedFields = getExcludedFromEqualsFor(parentCo) || [];
+    const excludedFields = getExcludedFromEqualsFor(childCo) || [];
+    excludedFromEquals.set(childCo, [...excludedFields, ...parentExcludedFields]);
+
+    const mappedFields = fieldsForEqualsMap.get(childCo) || new Map();
+    const parentMappedFields = fieldsForEqualsMap.get(parentCo) || new Map();
+    Array.from(parentMappedFields.keys())
+      .filter((key) => !Array.from(mappedFields.keys()).includes(key))
+      .forEach((key) => {
+        fieldsForEquals(...parentMappedFields.get(key))(new childCo(), key);
+      });
+  }
+}
+
+export function excludeFromEquals(object: any, propertyName: string): any {
```

Comment:
Please add Typedocs to this function.  It looks like this relates to the `@excludeFromEquals` annotations I'm seeing elsewhere, right?